### PR TITLE
Add Linting support

### PR DESCRIPTION
**Pull request by JeffreyMFarley**
Supersedes https://github.com/cfpb/ccdb5-ui/pull/32

Work towards #25 #26

## Additions

- Lint Rules

## Changes

- Lots

## Testing

- 💯 

## Todos

- Take care of the 175 warnings as modules are being worked

## Checklist

* [x] Changes are limited to a single goal (no scope creep)
* [x] Code can be automatically merged (no conflicts)
* [x] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [x] Passes all existing automated tests
* [x] New functions include new tests
* [x] New functions are documented (with a description, list of inputs, and expected output)
* [x] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### CODE REVIEW

**Comment by sebworks** [comment]
looks good, 👍

File: src/Filters/SingleCheckbox.jsx [hunk 1]
```diff
@@ -3,12 +3,12 @@ import React from 'react';
 export default class SingleCheckbox extends React.Component {
   render() {
     return (
-      <section className="single-checkbox">
```
Comment (sephcoster) [inline src/Filters/SingleCheckbox.jsx]: How do we feel about single checks for HTML / JSX? Seems like this is a case where you'd stick to double quotes for actual HTML properties but single quotes for JavaScript-related strings for readability but I'm OK either way as long as we're consistent.

File: src/constants/index.jsx [hunk 2]
```diff
@@ -37,17 +37,17 @@ export const knownFilters = [
...
-export const VK_UP                     = 38
+export const VK_DOWN = 40
+export const VK_ENTER = 14
+export const VK_ESCAPE = 27
+export const VK_RETURN = 13
+export const VK_TAB = 9
+export const VK_UP = 38
 
-///////////////////////////////////////////////////////////////////////////////
+/// ////////////////////////////////////////////////////////////////////////////
```
Comment (sephcoster) [inline src/constants/index.jsx]: Some weirdness here adding a space on the eslint fix but nothing to necessarily change. I also liked the formatting of constants but not necessarily worth writing a very specific lint rule for that spacing. Might be worth ignoring that file, however to allow for more readable constants if we prefer that format.

File: src/Filters/SingleCheckbox.jsx (hunk 1 above)
Comment (JeffreyMFarley) [inline src/Filters/SingleCheckbox.jsx]: I agree with you, but not sure how to configure eslint to do that

Comment (JeffreyMFarley) [inline src/Filters/SingleCheckbox.jsx]: I think I found where I can change the HTML/JSX quotes.  Will resubmit because I agree with ypu

**Pull request merged**